src/ipc/ipc-helper.c

```diff
@@ -205,7 +205,7 @@ int ipc_pipeline_complete(struct ipc *ipc, uint32_t comp_id)
 	/* check whether pipeline exists */
 	ipc_pipe = ipc_get_pipeline_by_id(ipc, comp_id);
 	if (!ipc_pipe) {
-		tr_err(&ipc_tr, "ipc: ipc_pipeline_complete looking for pipe component id %d failed",
+		tr_err(&ipc_tr, "ipc: ipc_pipeline_complete looking for pipe component id 0x%x failed",
```

Comment:
should we use `%#x`?

Reply:
@lyakh I think the problem has been that sof-logger/IPC3 doesn't handle this. We've had some usage (e.g. in pipeline-trace.h) but so far only in IPC4 only code.

That's also the reason why this cleanup has not been done yet in generic code, as you end up in some complicated compromises to make (as the many of these ids are bitmasks in IPC4 while there are unique integers in IPC3, so there's no single best represention of the numbers).

Reply:
Emphasis "I think". If "%#x" works, by all means let's go with that, but someone needs to ack it works with sof-logger.

Reply:
`%x` is already in use in many places:
```
git grep -E 'tr_.*%.{0,3}x'
git grep -E 'comp_.*%.{0,3}x'
```
If only one of these statements is a frequent one, then verifying it works it's just the matter of looking stable-v2.2 test results in any recent sof-test or linux Pull Request:

https://github.com/thesofproject/sof-test/pulls
https://github.com/thesofproject/linux/pulls

Reply:
>  If "%#x" works

BTW `%#x` is less obvious and readable than `0x%x` for near-zero advantage?

https://pubs.opengroup.org/onlinepubs/9699919799/functions/printf.htmlage